tests/descriptor_tests/test_shotgun.py

```diff
@@ -0,0 +1,454 @@
+# Copyright (c) 2016 Shotgun Software Inc.
```

Comment:
Tests ❤️ 

Would it be possible to add some testing for failed downloads or unzipping. This seems to be a recurring theme for clients and making sure we don't break that error reporting in the `__future__` would be nice.

Reply:
let me try to add some more tests.